api/internal/handler/data_loader/loader/openapi3/import.go

```diff
@@ -0,0 +1,180 @@
+				interestedAuthentication = true
+				interestedAuthenticationList[name] = security
+			}
+		}
+	}
+
+	// create upstream when servers field not empty
+	if len(s.Servers) > 0 {
+		var upstream entity.Upstream
+		upstream, globalPath = generateUpstreamByServers(s.Servers, globalUpstreamID)
+		data.Upstreams = append(data.Upstreams, upstream)
+	}
+
+	// each one will correspond to a route
+	for uri, v := range s.Paths {
+		// replace parameter in uri to wildcard
+		realUri := regURIVar.ReplaceAllString(uri, "*")
+		// generate route name
+		routeID := o.TaskName + "_" + strings.NewReplacer("/", "-", "{", "", "}", "").Replace(strings.TrimPrefix(uri, "/"))
+
+		// decide whether to merge multi-method routes based on configuration
+		if o.MergeMethod {
+			// create a single route for each path, merge all methods
+			route := generateBaseRoute(routeID, v.Summary)
+			route.Uris = []string{globalPath + realUri}
+			route.UpstreamID = globalUpstreamID
+			for method := range v.Operations() {
+				route.Methods = append(route.Methods, strings.ToUpper(method))
+			}
+			data.Routes = append(data.Routes, route)
+		} else {
+			// create routes for each method of each path
+			for method, operation := range v.Operations() {
+				subRouteID := routeID + "_" + method
+				route := generateBaseRoute(subRouteID, operation.Summary)
+				route.Uris = []string{globalPath + realUri}
+				route.Methods = []string{strings.ToUpper(method)}
+				route.UpstreamID = globalUpstreamID
+
+				// Processing plugins in non-method merge mode
+				// In method merge mode, different methods may have different authentication
+				// methods, so no plugins are attached to them.
+				if interestedAuthentication &&
+					operation.Security != nil &&
+					len(*operation.Security) > 0 {
+					attachAuthenticationPlugin(route, interestedAuthenticationList, operation.Security)
+				}
+				data.Routes = append(data.Routes, route)
+			}
+		}
+	}
+	return data, nil
+}
+
+// Adding authentication plugins to route based on the security scheme
+// Currently supported: apikey
+func attachAuthenticationPlugin(route entity.Route, refs map[string]*openapi3.SecuritySchemeRef, requirements *openapi3.SecurityRequirements) {
+	for _, requirement := range *requirements {
+		for name := range requirement {
+			// Only add plugins that we can currently handle
+			if _, ok := refs[name]; ok {
+				route.Plugins[authenticationMappings[name]] = map[string]interface{}{}
+			}
+		}
+	}
+}
+
+// Generate APISIX upstream from OpenAPI servers field
+// return upstream and uri prefix
+// Tips: It will use only the first server in servers array
+func generateUpstreamByServers(servers openapi3.Servers, upstreamID string) (entity.Upstream, string) {
+	upstream := entity.Upstream{
+		BaseInfo: entity.BaseInfo{ID: upstreamID},
+		UpstreamDef: entity.UpstreamDef{
+			Name: upstreamID,
+			Type: "roundrobin",
+		},
+	}
+
+	u, err := url.Parse(servers[0].URL)
```

Comment:
Only parse servers[0] ?

Reply:
This is based on the following considerations:

- The first servers are usually the node that is logically "used first".
- Different servers may have different global paths
```text
What's "global paths"?

servers:
- url: https://z.com/test/
- url: https://x.com/non-test/
- url: https://c.com/test-now/

According to the principle of APISIX, we need to set /test, /non-test,
/test-now, respectively, as the prefix of the routing uri of APISIX,
for example, set the original /user/* to /test/user/*, at this point
we can't operate.
```

Do you have any better suggestions?

Reply:
I don't think this is appropriate, it generates a lot of useless routes. 🤔

Reply:
Why the other routes are useless?

Reply:
> This means that for a URI, when we have multiple servers fields, it will generate at most (number of servers * number of paths * number of methods); for example with 5 servers, 10 paths, each with 3 methods, at most it will generate 5 * 10 * 3 = 150 routes, which I think is a bad use of experience

I think the servers in open API should be treated equally.
If the user doesn't want to use the server[0], he should change all the path in the route.It's also a bad experience.

Reply:
I am a little confused about `servers`, should they be the URL exposed by the gateway or the URL of the API backend service, please confirm, thanks.

Reply:
hi @bzp2010, please check this, it affects the overall convert

Reply:
> I am a little confused about servers, should they be the URL exposed by the gateway or the URL of the API backend service, please confirm, thanks.

@nic-chen We can look at it this way, during API debugging it is the address actually requested by the debugger, so it should be the address provided by APISIX to the outside world, and the original value should be upstream.

我们可以这样看待这个问题，在API调试过程中，它是调试器实际请求的地址，因此它应当是APISIX对外提供的地址，而原始值应该是上游的。

Reply:
@bzp2010 I suggest that this PR not deal with `servers` first, and deal with it later if there is a strong demand; instead of dealing with imperfections now.

Reply:
@nic-chen Well, I decided to remove this feature and replace it with a **default empty upstream** that the user can simply modify to change the upstream associated with all routes.